lib/ohai/plugins/cpu.rb

```diff
@@ -46,13 +47,211 @@ def parse_bsd_dmesg(&block)
     cpuinfo
   end
 
-  collect_data(:linux) do
+  def range_to_a(range)
+    range_array = []
+    range.split(",").each do |cpu|
+      if /\d+-\d+/.match?(cpu.to_s)
+        range_array << Range.new(*cpu.split("-").map(&:to_i)).to_a
+      else
+        range_array << cpu.to_i
+      end
+    end
+    range_array.flatten
```

Comment:
you can probably `range.split(",").each_with_object([]) { |cpu| if...else..end }.flatten` (only probably multiline)

Reply:
Am I just changing `.each` with `.each_with_object` or is there something more I need to do? If so, chefstyle complains if I do that.

Reply:
The only change here is that you don't have to pre-define range_array, but I think this is fine as-is

Reply:
So it could be this:
```
def range_to_a(range)
  range.split(",").each_with_object([]) do |cpu, arr|
    if /\d+-\d+/.match?(cpu.to_s)
      arr << Range.new(*cpu.split("-").map(&:to_i)).to_a
    else
      arr << cpu.to_i
    end
  end.flatten
end
```
is Lamont's point.

But also, this  comment is inaccurate. `range` is a thing in ruby and you can't pass this a range.  It takes a _string_ that looks like a range.